src/service/messageSubscriber/RetryFailureStrategy.ts

```diff
@@ -0,0 +1,18 @@
+import { IFailureStrategy } from './IFailureStrategy';
+import { SubscriptionMessages, SubscriptionMessagesConstructor } from './SubscriptionMessagesRepository';
+
+export class RetryFailureStrategy implements IFailureStrategy {
+  constructor(private retryTimes: number) {}
+
+  execute<T>(queueMessage: SubscriptionMessages<T>): SubscriptionMessagesConstructor<T> | null {
+    const retryTimesWhenCounterStartFrom0 = this.retryTimes - 1;
+    if (queueMessage.retries < retryTimesWhenCounterStartFrom0) {
+      return {
+        subscriptionId: queueMessage.subscriptionId,
```

Comment:
I guess here you can change the queue, right? For example for dead letter implementation?